esmvalcore/configuration/_config_validators.py

```diff
@@ -0,0 +1,298 @@
+
+
+def validate_bool(value, allow_none=False):
+    """Check if the value can be evaluate as a boolean."""
+    if (value is None) and allow_none:
+        return value
+    if not isinstance(value, bool):
+        raise ValueError(f"Could not convert `{value}` to `bool`")
+    return value
+
+
+def validate_path(value, allow_none=False):
+    """Return a path object."""
+    if (value is None) and allow_none:
+        return value
+    try:
+        path = Path(value).expanduser().absolute()
+    except TypeError as e:
+        raise ValueError(f"Expected a path, but got {value}") from e
+    else:
+        return path
+
+
+def validate_positive(value):
+    """Check if number is positive."""
+    if value < 0:
+        raise ValueError(f'Expected a positive number, but got {value}')
+    return value
+
+
+def _chain_validator(*funcs):
+    """Chain a series of validators."""
+    def chained(value):
+        for func in funcs:
+            value = func(value)
+        return value
+
+    return chained
+
+
+validate_string = _make_type_validator(str)
+validate_string_or_none = _make_type_validator(str, allow_none=True)
+validate_stringlist = _listify_validator(validate_string,
+                                         doc='return a list of strings')
+validate_int = _make_type_validator(int)
+validate_int_or_none = _make_type_validator(int, allow_none=True)
+validate_float = _make_type_validator(float)
+validate_float_or_none = _make_type_validator(float, allow_none=True)
+validate_floatlist = _listify_validator(validate_float,
+                                        doc='return a list of floats')
+
+validate_path_or_none = _make_type_validator(validate_path, allow_none=True)
+
+validate_pathlist = _listify_validator(validate_path,
+                                       doc='return a list of paths')
+
+validate_int_positive = _chain_validator(validate_int, validate_positive)
+validate_int_positive_or_none = _make_type_validator(validate_int_positive,
+                                                     allow_none=True)
+validate_dict = _make_type_validator(dict)
+
+
+def validate_positive(value):
+    """Reject negative values."""
+    if value is not None and value < 1:
+        raise ValueError(f"Must be larger be larger than 0. Got {value}.")
+    return value
+
+
+def validate_check_level(value):
+    from ..cmor.check import CheckLevels
+
+    if isinstance(value, str):
+        return CheckLevels[value.upper()]
+
+    return CheckLevels(value)
+
+
+def validate_diagnostics(diagnostics):
+    from .._recipe import TASKSEP
+
+    if isinstance(diagnostics, str):
+        diagnostics = diagnostics.split(' ')
+    return {
+        pattern if TASKSEP in pattern else pattern + TASKSEP + '*'
+        for pattern in diagnostics or ()
+    }
+
+
+class ESMValToolDeprecationWarning(UserWarning):
+    # Custom warning, because DeprecationWarning is hidden by default
+    pass
+
+
+def deprecate(func, variable, version: str = None):
```

Comment:
This package might also be nice, or do you think it's too much?
https://pypi.org/project/Deprecated/

Reply:
Seems like a bit much, and it solves the problem in the same way (using decorators). Is there anything that you think is missing from what we have now?

Reply:
What I liked about the package is that it supports mentioning stuff in the documentation

Reply:
This seems to be specific for functions. We are not really deprecating functions though, only keyword arguments.